enricher/api/src/main/java/io/fabric8/maven/enricher/api/AbstractHealthCheckEnricher.java

```diff
@@ -0,0 +1,90 @@
+package io.fabric8.maven.enricher.api;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+
+/**
+ * Enriches containers with health check probes.
+ */
+public abstract class AbstractHealthCheckEnricher extends BaseEnricher {
+
+    public AbstractHealthCheckEnricher(EnricherContext buildContext, String name) {
+        super(buildContext, name);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            if (item instanceof Deployment) {
+                Deployment deployment = (Deployment) item;
+                Container container = getContainer(deployment);
+                if (container != null) {
+                    if (container.getReadinessProbe() == null) {
+                        Probe probe = getReadinessProbe();
+                        if (probe != null) {
+                            log.info("Adding readiness " + describe(probe));
+                            container.setReadinessProbe(probe);
+                        }
+                    }
+
+                    if (container.getLivenessProbe() == null) {
+                        Probe probe = getLivenessProbe();
+                        if (probe != null) {
+                            log.info("Adding liveness " + describe(probe));
+                            container.setLivenessProbe(probe);
+                        }
+                    }
+                }
+            }
+        }
+        builder.withItems(items);
+    }
+
+    private Container getContainer(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers().size() > 0) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            return containers.get(containers.size() - 1); // last one
```

Comment:
You are probably right, it should be the enricher which runs after the generator who is completely responsible for the health checks (at the end is a k8s / openshift concept. However, ill be funny how to deal with the new [Docker Healthchecks](https://docs.docker.com/engine/reference/builder/#/healthcheck) ;-)

Also, not every image is defined by a `Generator`. However the problem what I see is, that an `Enricher` might not have enough information about the images it contains.

So I wonder whether we should extend the `Enricher` API to also get hold of the `ImageConfiguration`s defining the image ? From this the image itself can be deeply inspected like e.g the Docker `HEALTHCHECKS`, exposed ports etc. 

Another idea: To have a generic health-check enricher which checks for an Image Label (or the mentioned Docker HEALTH_CHECKS). This generic health-check enricher is independent of spring-boot, wildfly-swarm etc.

Then:

* spring-boot and widlfly-swarm generator could define the specific health check info (Docker Health Check or Labels or Envs)
* Generic images can create the appropriate info on their own by creating their own labels in the format that the generic enricher understands.

Maybe we can settle on Docker HEALTH_CHECK as this 'marker' as is sufficient generic ? For this to work:

* d-m-p must be improved to support health checks in `ImageConfiguration` (and must take care about backwards compatibility as Docker health checks are quite new).
* `Enricher` must be enhanced to get the list of ImageConfigurations.
* A generic `HealthCheckEnricher` must be written.

That way a health check enricher would only kick in if an image defines a specific health-check, so it's also quite flexible (and close the original app) and might be useful for the image per-se, even when not used in a K8s / OpenShift context.

Still thinking loud a bit. One problem could be that, Docker health_checks are so new that the image format is not supported by K8s or OpenShift. If so, we can still work with a similar label.